src/App.jsx

```diff
@@ -1,26 +1,195 @@
 import { useState } from 'react'
+import { useEffect } from 'react'
 import './App.css'
 import './stylesheets/movielist.css'
 
 import MovieList from './components/MovieList'
 import Header from './components/Header'
 import Footer from './components/Footer'
 
+const baseDetailsUrl = 'https://api.themoviedb.org/3/movie/now_playing?language=en-US'; //url that we are sending fetch request to that returns array with movies now playing and details without page at end
+const baseSearchUrl = 'https://api.themoviedb.org/3/search/movie'; //url that is the request that returns movie results depending on a query parameter which we are getting from our SearchBar component
 
 const App = () => {
+
+  const [movieList, setMovieList] = useState([]); //decalre a movieList as an empty array. We will populate it with a fetch request to imdb API
+  const [pageNum, setPageNum] = useState(1); //pageNum will keep track of the movies we are rendering. Whenever we click the load more button the page increases and we request next page of movies
```

Comment:
Looking at the logic, I think will be better keep this as useRef instead. The difference is that updating a state variable will make the dom tree to be refreshed, and in this case we just want to update the dom when the movieList is updated.